opsmngr/diagnostics.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	diagnosticsBasePath = "groups/%s/diagnostics"
+)
+
+// DiagnosticsService is an interface for interfacing with Diagnostic Archives in MongoDB Ops Manager APIs
+// https://docs.opsmanager.mongodb.com/current/reference/api/diagnostic-archives/
+type DiagnosticsService interface {
+	Get(context.Context, string, io.Writer) (*atlas.Response, error)
+}
+
+type DiagnosticsServiceOp struct {
+	Client atlas.GZipRequestDoer
```

Comment:
We have to change GZipRequestDoer to accept a body if we want to allow the client to set the query params `limit` and `minutes`.

Reply:
Do we want to do it as part of this PR? @gssbzn